src/Test.Utilities/CodeFixTestBase.cs

```diff
@@ -128,10 +158,22 @@ private void VerifyFixAll(string language, DiagnosticAnalyzer analyzerOpt, CodeF
                     throw new Exception($"Unable to invoke code fix at index '{codeFixIndex.Value}', only '{actions.Count}' code fixes were registered.");
                 }
 
-                document = document.Apply(actions.ElementAt(codeFixIndex.Value));
+                var codeAction = actions.ElementAt(codeFixIndex.Value);
+
+                if (!testFixAll)
+                {
+                    document = document.Apply(codeAction);
+                }
+                else
+                {
+                    string diagnosticIdToFix = triggerDiagnostic.Id;
+                    var diagnosticsToFix = fixableDiagnostics.Where(d => d.Id == diagnosticIdToFix);
+                    document = ApplyFixAll(document, codeAction, codeFixProvider, codeFixProvider.GetFixAllProvider(), diagnosticIdToFix, diagnosticsToFix);
```

Comment:
>document = ApplyFixAll(document, codeAction, codeFixProvider, codeFixProvider.GetFixAllProvider(), diagnosticIdToFix, diagnosticsToFix); [](start = 20, length = 136)

Location of the call means that we will make it within the while loop. And can repeat it again and again. 

If we have an error within a fix all algorithm for a given analyzer that fixes, for example, just a single error not all; this fixer would pass all tests because it allows fixing one by one.

Reply:
Thanks, as we discussed, I will modify this to do a single iteration. For tests which are relying on iterative fix/diagnostic/fix output, they can be handled specially (call separate API for verification)

Reply:
Fix All does not always fix all in a single iteration. For StyleCop Analyzers we fail the test if it fails to fix everything in one iteration, but have a parameter which allows you to change the behavior. Most of the work to implement this was here:

https://github.com/DotNetAnalyzers/StyleCopAnalyzers/pull/1989

Reply:
Thanks Sam. After discussing with Ivan, we do something similar for the default case. We only run one FixAll iteration and fail it doesn't fix everything. Tests which fail this way can turn off default FixAll validation with an optional flag, and have a separate VerifyFixAll invocation with appropriate fixedSource.